### PR TITLE
bug 1774203: document uploading to stage

### PR DESCRIPTION
**Pull request by willkg**
This adds some documentation to the upload API encouraging people to use
the stage environment for testing symbol uploads and walks through how
to do that.

### CODE REVIEW

**Comment by willkg** [comment]
@edugfilho Can you skim this and see if it makes sense to you? The gist of it is that we have two environments and we want to document enough of the stage environment and how to use it that people who are testing uploads and testing scripts have enough information here to understand what we support and do what they need to do.

File: docs/conf.py [hunk 1]
```diff
@@ -96,7 +96,7 @@
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+language = "en"
```
Comment (willkg) [inline docs/conf.py:99]: Sphinx 5 raises a warning with our previous value.

File: docs/upload.rst [hunk 2]
```diff
@@ -73,17 +74,6 @@ For example, here's the contents of a symbols ZIP file from a Firefox build::
...
 
-Permissions and auth token
---------------------------
-
-Uploading requires two things:
-
-1. an account with upload permissions
-2. an auth token for that account with upload permissions
-
-The auth token is used in the HTTP POST to authenticate the upload.
```
Comment (willkg) [inline docs/upload.rst:84]: This part was in this document twice, so I nixed it from here figuring the other section is more fleshed out.

**Review by edugfilho — APPROVED** [review]
looks great!

**Comment by willkg** [comment]
Thank you!

**Pull request merged**